lib/runtime-core/src/codegen.rs

```diff
@@ -274,7 +273,8 @@ fn requires_pre_validation(backend: Backend) -> bool {
 
 /// A sink for parse events.
 pub struct EventSink<'a, 'b> {
-    buffer: SmallVec<[Event<'a, 'b>; 2]>,
+    /// document this
+    pub buffer: Vec<Event<'a, 'b>>,
```

Comment:
We can talk about this later, but it may make sense to keep using a SmallVec, but maybe a larger SmallVec depending on how everything works; I'm mostly reviewing for Rust style right now so I didn't catch the event sink behavior

Reply:
Sure. I am ambivalent about what vector is used as I am still mostly ignorant to the subtle differences.

Reply:
In this case it's a very small thing, SmallVec is just a combination of static array and vector, so that you can avoid heap allocations in most cases while still supporting outliers that need an arbitrarily large vector.

It's especially useful if you're writing a generic data type which under normal circumstances only needs to hold a little bit of data, but it's possible that it could sometimes hold a lot or that some other code could use it in ways you didn't anticipate.  I'm not sure useful it is here 🤷‍♂